nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,326 @@
+        final String linuxPrimaryKey = context.getProperty(LINUX_PRIMARY_KEY).evaluateAttributeExpressions().getValue();
+        final boolean jvmMetricsCollected = context.getProperty(SEND_JVM_METRICS).asBoolean();
+
+        String logName = context.getProperty(LOG_ANALYTICS_CUSTOM_LOG_NAME).evaluateAttributeExpressions().getValue();
+        final String instanceId = context.getProperty(INSTANCE_ID).evaluateAttributeExpressions().getValue();
+        String groupIds = context.getProperty(PROCESS_GROUP_IDS).evaluateAttributeExpressions().getValue();
+        try {
+            if(groupIds == null) {
+                ProcessGroupStatus status = context.getEventAccess().getControllerStatus();
+                String processGroupName = status.getName();
+                List<Metric> allMetrics = collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected);
+                sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+            } else {
+                if (!groupIds.contains(",")) {
+                    ProcessGroupStatus status = context.getEventAccess().getGroupStatus(groupIds.trim());
+                    String processGroupName = status.getName();
+                    List<Metric> allMetrics = collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected);
+                    sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+                } else {
+                    List<Metric> allMetrics = new ArrayList<>();
+                    for(String groupId: groupIds.split(",")) {
+                        groupId =groupId.trim();
+                        ProcessGroupStatus status = context.getEventAccess().getGroupStatus(groupId);
+                        String processGroupName = status.getName();
+                        allMetrics.addAll(collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected));
+                    }
+                    sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+                }
+            }
+        } catch (IOException | IllegalArgumentException e) {
+            getLogger().error("Exception in outmost block", e);
+        }
+    }
+
+    /**
+     *  Construct HttpsURLConnection and return it
+     * @param workspaceId your azure log analytics workspace id
+     * @param logName log table name where metrics will be pushed
+     * @return HttpsURLConnection to your azure log analytics workspace
+     * @throws IOException when there is an error in creating https url connection with workspace id
+     */
+    protected HttpsURLConnection getHttpsURLConnection(final String workspaceId, final String logName)
+        throws IOException {
+        String dataCollectorEndpoint =
+            MessageFormat.format("https://{0}.ods.opinsights.azure.com/api/logs?api-version=2016-04-01", workspaceId);
+        URL url = new URL(dataCollectorEndpoint);
+        HttpsURLConnection conn = (HttpsURLConnection) url.openConnection();
```

Comment:
minor: Not sure if OkHttpClient provides value but that is used in InvokeHTTP and provides a cleaner interface than using the connection directly.

Reply:
I see your point. But, not sure whether we bring value with adding this dependency on OkHttpClient, given we only post json data and set authorization property.  I will create a separate branch and see any regression issues later.

Reply:
I have replaced HttpsURLConnection with Apache HttpClient, after evaluating jersey-client, okhttp, and apache httpclient.